cmd/telegraf/telegraf.go

```diff
@@ -256,16 +256,6 @@ func runAgent(ctx context.Context,
 
 	logger.SetupLogging(logConfig)
 
-	if *fRunOnce {
-		wait := time.Duration(*fTestWait) * time.Second
-		return ag.Once(ctx, wait)
-	}
-
-	if *fTest || *fTestWait != 0 {
-		wait := time.Duration(*fTestWait) * time.Second
-		return ag.Test(ctx, wait)
-	}
-
 	log.Printf("I! Loaded inputs: %s", strings.Join(c.InputNames(), " "))
```

Comment:
This will also, unfortunately, print out the "loaded outputs", which may lead to even more confusion about why outputs are not run doing `--test` or `--once`. What if we only printed the inputs?

Reply:
Outputs are definitely run with `--once`, but question is, are they loaded?

Reply:
Based on https://github.com/influxdata/telegraf/blob/52c422af3b32ab4185f82ce3f7789255934ab4b8/cmd/telegraf/telegraf.go#L224-L226 I would say no.

Reply:
So I stand corrected on `--once`, but they do show up and they should *not* with `--test` with this PR.

Reply:
That's just what I'm saying. I don't believe there are any outputs loaded in that case.

Reply:
I can guard the output with an if, but this would make it harder to read...

Reply:
Any preference?

Reply:
It seems they are loaded, in that case i would not print that log message for outputs for `--test`.

Reply:
What about doing the following:

- Only move the ag.Once block down, since we do want to see everything that is loaded
- Leave ag.Test up above all the log/prints, and add a log printf statement that shows what inputs were loaded

Reply:
Test also does processors (and aggregators?) But I agree on moving only the once block.

Reply:
I've added a proposal. We only print the loaded outputs in 'non-testing' mode. Otherwise we show a big red warning that outputs are not used in testing mode (see #10080). How does that sound to you?